### DISCUSSION TITLE
Event vs Trace

### DISCUSSION THREAD

**Discussion by MacNale** [weaveworks-experiments/kspan#34] [General]
Thank you for the idea. Learned a lot.

Wondering if we are trying to force an "event" to act as "trace" and is it a good idea? That led me to read about "events" in OpenTelemetry. It looks like, Opentelemetry recognizes Trace, Logs, Metrics as fundamental tenants of observability. There is no mention of events except as a part of span.  Each openTelemetry span contains metadata about the operation, such as its name, start and end timestamps, attributes, **events**, and status. 

For now kspan works. However, I wonder if Kubernatives should emit opentelemetry natively.  Does it already?

**Comment by bboreham** [comment]
I believe as of today Kubernetes does not emit any traces.
There are a couple of enhancement proposals: [tracing calls to api-server](https://github.com/kubernetes/enhancements/issues/647) and [passing context through objects](https://github.com/kubernetes/enhancements/pull/2312).

> is it a good idea

It's kind of a hack, but I love how well it works!

**Reply by MacNale** [reply]
https://github.com/kubernetes/kubernetes/pull/94942
